sync/syncer.go

```diff
@@ -82,15 +82,20 @@ type Syncer struct {
 	clock             timesync.LayerTimer
 	exit              chan struct{}
 	currentLayerMutex sync.RWMutex
+	syncFuncWg        sync.WaitGroup
 }
 
 func (s *Syncer) ForceSync() {
 	s.forceSync <- true
 }
 
 func (s *Syncer) Close() {
+	s.Info("Closing syncer")
 	close(s.exit)
 	close(s.forceSync)
+	// TODO: broadly implement a better mechanism for shutdown
+	time.Sleep(5 * time.Millisecond) // "ensures" no more sync routines can be created, ok for now
```

Comment:
do you really need this sleep ?

Reply:
Yes and no. It probably won't happen but there is a potential race there (what I explained to you offline). The same race exists in the block_listener. Since it is just shutdown and since we probably should think of a better way to handle shutdowns I thought that a 5ms delay is good enough for now.